pkg/api/datasources.go

```diff
@@ -248,6 +248,11 @@ func (hs *HTTPServer) UpdateDataSource(c *models.ReqContext) response.Response {
 	datasourcesLogger.Debug("Received command to update data source", "url", cmd.Url)
 	cmd.OrgId = c.OrgId
 	cmd.Id = c.ParamsInt64(":id")
+
+	if cmd.Url == "" {
+		datasourcesLogger.Error("Received empty data source URL as part of data source command")
+		return response.Error(400, "Validation error, empty URL", nil)
+	}
```

Comment:
I assume you will move this to validateUrl after UI is fixed..

Reply:
Exactly